### PR TITLE
[feature/dashboard-oop] Publish dashboard for specific RIDs before packing

### PR DESCRIPTION
**Pull request by danegsta**
This PR publishes the Dashboard for every RID we support in the Aspire workload before building the workload pack.
 
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1771)

### CODE REVIEW

File: src/Aspire.Dashboard/Aspire.Dashboard.csproj [hunk 1]
```diff
@@ -1,24 +1,15 @@
...
 
-  <PropertyGroup>
-    <Description>Dashboard browser interface for .NET Aspire.</Description>
-    <PackageLicenseExpression>MIT</PackageLicenseExpression>
-    <Title>.NET Aspire Dashboard</Title>
-    <PackageProjectUrl>https://github.com/dotnet/aspire</PackageProjectUrl>
-    <PackageIcon>Aspire_icon_256.png</PackageIcon>
-    <RepositoryType>git</RepositoryType>
-    <RepositoryUrl>https://github.com/dotnet/aspire</RepositoryUrl>
-  </PropertyGroup>
```
Comment (drewnoakes) [inline src/Aspire.Dashboard/Aspire.Dashboard.csproj:20]: If these props are being removed, please also remove the `None` item for the icon below too.

File: eng/workloads/workloads.csproj [hunk 2]
```diff
@@ -36,15 +36,16 @@
     <PackageReference Include="Microsoft.DotNet.Build.Tasks.Installers" GeneratePathProperty="true" />
   </ItemGroup>
 
-  <!-- The list of runtimes DCP provides binaries for -->
+  <!-- The list of supported runtimes -->
```
Comment (drewnoakes) [inline eng/workloads/workloads.csproj:39]: I find it helpful to know that this list is synchronized with a similar list in DCP.

```suggestion
  <!-- The list of supported runtimes, which matches those of DCP -->
```

**Review by drewnoakes — COMMENTED** [review]
Looks good to me, but I'm not an expert.

**Review by drewnoakes — APPROVED** [review]
Oops, meant to approve in my last review.

**Review by joperezr — APPROVED** [review]
Awesome, thanks so much for taking care of this! I actually needed this for the container image, so I may just use these packs to get the published output from here.

File: eng/workloads/workloads.csproj [hunk 3]
```diff
@@ -74,8 +75,8 @@
       </ShortNames>
     </ItemGroup>
 
-    <MSBuild Projects="../dashboardpack/dashboardpack.csproj" Targets="restore;build" Properties="PackageOutputPath=$(PackageSource)" />
-    <MSBuild Projects="../dcppack/dcppack.csproj" Targets="restore;build" Properties="DcpRuntime=%(_DcpRuntimes.Identity);PackageOutputPath=$(PackageSource)" />
+    <MSBuild Projects="../dashboardpack/dashboardpack.csproj" Targets="restore;build" Properties="DashboardRuntime=%(_PackRuntimes.Identity);PackageOutputPath=$(PackageSource)" />
```
Comment (joperezr) [inline eng/workloads/workloads.csproj:78]: Wait, don't we need to also call the publish target here?

Comment (joperezr) [reply eng/workloads/workloads.csproj:78]: I don't think it is actually getting packed though, at least not locally so I may be doing something wrong. I pulled your branch and this is what I see:

image

No tools folder.

Comment (joperezr) [reply eng/workloads/workloads.csproj:78]: I just downloaded the one produced from your build, and that one doesn't have it either.

Comment (joperezr) [reply eng/workloads/workloads.csproj:78]: I think I found the issue. It is a timing thing, we are defining None item statically in the dashboardpacks project, but the actual published assets don't get emitted until the BeforeBuild target is called. I will put up a PR to fix this.

Comment (danegsta) [reply eng/workloads/workloads.csproj:78]: Good catch; it must have seemed like it was working locally for me due to a cached build from a previous run

Comment (joperezr) [reply eng/workloads/workloads.csproj:78]: Yup. Put up PR https://github.com/dotnet/aspire/pull/1796 to fix this.

**Pull request merged**